libdrgn/dwarf_info_cache.c

```diff
@@ -1475,6 +1485,52 @@ drgn_dwarf_object_find(const char *name, size_t name_len, const char *filename,
 	return &drgn_not_found;
 }
 
+struct drgn_error *
+drgn_dwarf_object_find_with_namespaces(const char *name, size_t name_len, const char *filename,
+				       enum drgn_find_object_flags flags,
+				       struct drgn_dwarf_index_namespace* ns,
+				       void *arg, struct drgn_object *ret) {
+	struct drgn_error *err;
+	struct drgn_dwarf_info_cache *dicache = arg;
+	char* ns_ptr;
+
+	if (!ns)
+		ns = &dicache->dindex.ns;
+
+find_ns:
+	ns_ptr = strstr(name, "::");
```

Comment:
`name` has a given length and isn't necessarily null-terminated, so this needs to use `memmem()`.